src/util/time.cpp

```diff
@@ -41,6 +42,22 @@ std::chrono::seconds GetMockTime()
     return g_mock_time.load(std::memory_order_relaxed);
 }
 
+NodeSteadyClock::time_point NodeSteadyClock::now() noexcept
+{
+    const auto mocktime{g_mock_steady_time.load(std::memory_order_relaxed)};
+    const auto ret{
+        mocktime.count() ?
+            mocktime :
+            std::chrono::steady_clock::now().time_since_epoch()};
+    return time_point{ret};
+};
+
+void SetMockSteadyTime(std::chrono::milliseconds mock_time_in)
+{
+    Assert(mock_time_in >= 0s);
+    g_mock_steady_time.store(mock_time_in, std::memory_order_relaxed);
```

Comment:
My approach for this would have been to reuse `SetMockTime`, which would set `g_mock_time` to the requested mocktime and `g_mock_steady_time` to `max(mock_time_in, g_mock_steady_time)` (to ensure it only ever increases).

The benefit of that approach would be that all existing tests that make use of mock time would now also mock steady time (assuming `NodeSteadyClock` is used throughout). 

I never implemented and tested this but I was curious if you considered this as well?

Reply:
i don't know.

i like being able to mock them seperately, and millisecond resolultion instead of second resolution is kind of nice here because it tends to be used for deadlines and timeouts, seconds are kind of clunky there.

also conceptually i think "`max(mock_time_in, g_mock_steady_time)`" is a wrong representation of how monotonic time works, it's basically a CPU cycle counter, it isn't linked to wall time at all.

Reply:
> in my current tests i in fact reset it in every test case.

Right, this wouldn't work with my approach as this is a global.